compiler/src/test/java/com/squareup/anvil/compiler/dagger/BindsMethodValidatorTest.kt

```diff
@@ -164,6 +194,30 @@ class BindsMethodValidatorTest(
     }
   }
 
+  @Test
+  fun `an extension function binding is valid`() {
+    compile(
+      """
+      package com.squareup.test
+ 
+      import dagger.Binds
+      import dagger.Module
+      import javax.inject.Inject
+
+      class Foo @Inject constructor() : Bar
+      interface Bar
+
+      @Module
+      abstract class BarModule {
+        @Binds
+        abstract fun Foo.bindsBar(): Bar
+      }
+      """
+    ) {
+      assertThat(exitCode).isEqualTo(OK)
```

Comment:
I'm not sure what you're testing here. Remove Anvil and the Dagger annotation processor from the build and the test would be green no matter what. Do you want to add a Dagger component to verify the binding method works?

Also, Anvil isn't doing anything with binding methods. There's also no Anvil annotation present in this entire test. It's a no-op at the moment.

Reply:
This is testing that our @Binds validation logic doesn't fail for a valid use-case; without the changes to `BindsMethodValidator` in this PR, this test fails because we would see that there was no parameter and throw an exception, while Dagger would compile successfully. Does that make sense, or do you think this is not worth including? For the purposes of this test it doesn't seem like including a Dagger component is necessary ; or at least, it's not clear why that would be needed here but not the other test cases. Open to being convinced otherwise :)

Reply:
I guess it doesn't hurt. I suggested to add a component and make sure that the binding works, to give this test more meaning in the sense: hey, here's a binding method and it works at runtime with Anvil or Dagger. I've done this at a few other places.

Reply:
Gotcha, that makes sense; I went ahead and added it to help with that clarity.